go/vt/vtgate/vcursor_impl_test.go

```diff
@@ -260,26 +260,37 @@ func TestSetTarget(t *testing.T) {
 func TestKeyForPlan(t *testing.T) {
 	type testCase struct {
 		vschema               *vindexes.VSchema
+		fkState               sqlparser.FkChecksState
 		targetString          string
 		expectedPlanPrefixKey string
 	}
 
 	tests := []testCase{{
 		vschema:               vschemaWith1KS,
 		targetString:          "",
-		expectedPlanPrefixKey: "ks1@primary+Collate:utf8mb4_0900_ai_ci+Query:SELECT 1",
+		expectedPlanPrefixKey: "ks1@primary+Collate:utf8mb4_0900_ai_ci+fkChecksState:+Query:SELECT 1",
 	}, {
 		vschema:               vschemaWith1KS,
 		targetString:          "ks1@replica",
-		expectedPlanPrefixKey: "ks1@replica+Collate:utf8mb4_0900_ai_ci+Query:SELECT 1",
+		expectedPlanPrefixKey: "ks1@replica+Collate:utf8mb4_0900_ai_ci+fkChecksState:+Query:SELECT 1",
```

Comment:
This is kind of weird. If `FkChecksState` is not set, we don't want to emit a `fkCheckState:` part of the key with no right-hand value. We can just skip it altogether.

Reply:
I agree with it.

Reply:
Fixed it.